### PR TITLE
Document service account configuration

### PR DESCRIPTION
**Pull request by sbryzak**
This is a work in progress, so far just experimenting with including screenshots / links etc in the docs.  Bulk of the actual content to follow.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-177-1`

**Comment by sbryzak** [comment]
Link to issue: #176 

This covers the basics for service accounts, I'll continue on by documenting service account authentication as part of #178

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-177-2`

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-177-3`

File: docs/source/developer.adoc [hunk 1]
```diff
@@ -1,3 +1,60 @@
...
+. Create a new branch representing the issue you will work on (*git checkout -b ISSUE-XXX*)
+. Make your changes, commit them to the issue branch
+. Push the changes to your forked repository
+. In Github, browse to the issue branch in your forked repository
+. Create a pull request for your issue branch, making sure to clearly link the issue that the PR is for
+. Update the issue with a link to the pull request
+
+== Service Accounts
+
+Service accounts are used by resource servers to interact with the fabric8-auth protection API.  This API requires a PAT (Protection API Token) in order for the resource servers to register newly created resources, and query the access rights for clients attempting to access resources that they don't own.
```
Comment (alexeykazakov) [inline docs/source/developer.adoc]: Resource server to AuthZ server is just one case of how service accounts can be used. SAs are more general. They can be used for any secured service-to-service communications.

File: docs/source/developer.adoc [hunk 2]
```diff
@@ -1,3 +1,60 @@
...
+
+Service account configuration is expected to be found in */etc/fabric8/service-account-secrets.conf*, in JSON format.  Here's an example configuration:
+
+[source,json]
+{
+    "accounts": [
+        {
+            "name":"fabric8-wit",
+            "id":"5dec5fdb-09e3-4453-b73f-5c828832b28e",
+            "secrets":["witsecret"]
```
Comment (alexeykazakov) [inline docs/source/developer.adoc]: Let's use hashes in the example. Plain text secrets won't work. We expect a bcrypt hash in the config file.

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-177-5`

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-177-6`

**Comment by alexeykazakov** [comment]
Replaced by https://github.com/fabric8-services/fabric8-auth/pull/187

**Comment by fabric8cd** [comment]
@sbryzak snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-177-7`

**Pull request closed without merging**